src/rust/engine/process_execution/src/nailgun/nailgun_pool.rs

```diff
@@ -497,7 +497,10 @@ impl Drop for BorrowedNailgunProcess {
         "Killing nailgun process {:?} due to cancellation.",
         process.as_ref().unwrap().name
       );
-      let _ = process.as_mut().unwrap().handle.kill();
+      if process.as_mut().unwrap().handle.kill().is_ok() {
+        // NB: This is blocking, but should be a short wait in general.
+        let _ = process.as_mut().unwrap().handle.wait();
```

Comment:
Can this hang?

Reply:
Not in my experience. Processes cannot actually handle `SIGKILL`, so you're really waiting for the OS to tear down the process in this case. That should only hang in case of an OS bug afaik.